### PR TITLE
app: add market list filter

### PR DESCRIPTION
**Pull request by buck54321**
This adds a small input at the top of the markets list that allows filtering the displayed markets.

The market list used to be populated in the html template, and the `MarketsPage` class would read its initial market list from there. To be in line with patterns used elsewhere, the market list is now
populated by javascript using js template elements pulled and cloned from the DOM.

### CODE REVIEW

**Comment by chappjc** [comment]
LGTM.  @itswisdomagain, wanna have a look?

File: client/webserver/site/src/js/markets.js [hunk 1]
```diff
@@ -1075,3 +1231,22 @@ function updateUserOrderRow (tr, order) {
...
+  })
+}
+
+/*
+ * logoSrc constructs the src attribute value for an img element that displays
+ * an asset's logo.
+ */
+function logoSrc (symbol) {
+  return `/img/coins/${symbol.toLowerCase()}.png`
+}
```
Comment (itswisdomagain) [inline client/webserver/site/src/js/markets.js:1252]: doc.js has a similar `logoPath (symbol)` function
https://github.com/decred/dcrdex/blob/0662a9f526aa4bc85347d436302e8ed32649ed77/client/webserver/site/src/js/doc.js#L118-L121

File: client/webserver/site/src/js/markets.js [hunk 2]
```diff
@@ -988,6 +969,181 @@ export default class MarketsPage extends BasePage {
...
+
+/*
+ * MarketList respresents the list of exchanges and markets on the left side of
+ * markets view. The MarketList provides utilities for adjusting the visibility
+ * and sort order of markets.
+ */
+class MarketList {
+  constructor (col) {
+    this.column = col
```
Comment (itswisdomagain) [inline client/webserver/site/src/js/markets.js:987]: `col`/`column` sounds like integer values to me (as in row/column). Maybe `listElement` or `listEl`?

File: client/webserver/site/src/js/markets.js [hunk 3]
```diff
@@ -988,6 +969,181 @@ export default class MarketsPage extends BasePage {
...
+    this.selected = null
+    const tmpl = tmplElement(col, 'xc')
```
Comment (itswisdomagain) [inline client/webserver/site/src/js/markets.js:989]: `exchangeTmpl` or `xcTmpl`

File: client/webserver/site/src/js/markets.js [hunk 4]
```diff
@@ -988,6 +969,181 @@ export default class MarketsPage extends BasePage {
...
+ */
+class ExchangeSection {
+  constructor (tmpl, dex) {
+    this.dex = dex
+    this.host = dex.host
+    const box = tmpl.cloneNode(true)
+    this.box = box
+    const header = tmplElement(box, 'header')
+    this.connectedIcon = tmplElement(header, 'disconnected')
+    if (dex.connected) Doc.hide(this.connectedIcon)
```
Comment (itswisdomagain) [inline client/webserver/site/src/js/markets.js:1071]: `connectedIcon` -> `disconnectedIcon`
Wondered for a minute why we're hiding the `connectedIcon` if the dex is connected, before seeing `tmplElement(header, 'disconnected')`

File: client/webserver/site/src/js/markets.js [hunk 5]
```diff
@@ -988,6 +969,181 @@ export default class MarketsPage extends BasePage {
...
+    this.selected = this.xcSection(host).marketRow(baseID, quoteID)
+    this.selected.row.classList.add('selected')
+  }
+
+  /* setConnectionStatus sets the visiblity of the disconnected icon based
+   * on the core.EpochNotification.
+   */
+  setConnectionStatus (note) {
+    this.xcSection(note.host).setConnected(note.connected)
+  }
```
Comment (itswisdomagain) [inline client/webserver/site/src/js/markets.js:1048]: `core.EpochNotification` does not contain `connected` prop, should be `core.ConnEventNote`

File: client/webserver/site/src/js/markets.js (hunk 4 above)
Comment (chappjc) [reply client/webserver/site/src/js/markets.js:1071]: I was ok with seeing it all the time, but the absence of an icon for connected with just a disconnected icon was the intended behavior.  Keeping clutter off the UI is a good idea though.

**Pull request merged**